---
title: Differential gene expression of human cardiocascular endothelial cells after different diets
output:
html_document: default
pdf_document: default
---
  
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## 1. Quality assessment

After sequencing, the quality of the FASTQ files was tested with FASTQC (scripts attached). Subsequently, the reads were trimmed to a minimal length of 75bp and by quality of the bases, using the tool trimmomatic (see trimmomatic log files and trimmomatic_FASTQC fiels attached). To subtract last traced of rRNA, the reads were aligned to different rRNA databases, excluding reads of rRNA sequencing from downstream analysis. This step was accomplished using the tool sortmerna (see sortmerna BASH script, log files and sortmerna_FASTQC files attached).


## 2. Aligning with salmon

After QA of the reads, the Pseudo-aligner "Salmon" was used to create count matrices. See code for this process attached (Pseudo_alignment script 1-3).


```{r, include = FALSE }

suppressPackageStartupMessages(library(tximport))
suppressPackageStartupMessages(library(rjson))
suppressPackageStartupMessages(library(DESeq2))
suppressPackageStartupMessages(library(hexbin))
suppressPackageStartupMessages(library(vsn))
suppressPackageStartupMessages(library(SummarizedExperiment))

```



## 4. Differential gene expression

For the differential gene expression analysis, we use DESeq2 and the edgeR package. We therefore  first load the data.

```{r, echo=FALSE, include = FALSE}

datadir <- "S:/Arbeitsgruppen VIRO/AG_Sch/Johannes Ptok_JoPt/WHK/DEG/Salmon/quant_data"

tx2gene <-  read.csv2(file.path(datadir, "Gene_nametx2gene.csv"))
files <- list.files(datadir, recursive = TRUE, pattern = "quant.sf",
                    full.names = TRUE)

samples <- read.csv2("S:/Arbeitsgruppen VIRO/AG_Sch/Johannes Ptok_JoPt/Transkriptomdaten/Endothelzellen/endo_sample_codes.csv")

x <- basename(gsub(".quant.sf", "", files))
names(files) <- x

txi_salmon <- tximport(files = files, type = "salmon", txOut = FALSE, tx2gene = tx2gene)

names(txi_salmon)
counts_salmon <- txi_salmon$counts
head(counts_salmon)
dim(counts_salmon)
counts_salmon <- round(counts_salmon)

```

```{r, echo=FALSE }

head(counts_salmon)

```

Next, we create the meta table for the analysis, describing the samples conditions.

```{r, echo=FALSE, include = FALSE}

samples <- read.csv2("S:/Arbeitsgruppen VIRO/AG_Sch/Johannes Ptok_JoPt/Transkriptomdaten/Endothelzellen/endo_sample_codes.csv")

id <- colnames(counts_salmon)
flow   <- samples$Flowcell[match(colnames(counts_salmon),samples$Samplename )]
group   <- samples$Group[match(colnames(counts_salmon),samples$Samplename )]



meta <- data.frame( id=id,
                    flow=flow,
                    group=group)


rownames(meta) <- meta$id

```

```{r, echo=FALSE }

head(meta)

```


We use the meta table and the count matrix for the creation of the "DESeqDataSet" object, needed for the analysis with DESEq2.

```{r, echo=FALSE, include = FALSE }

stopifnot(all(colnames(counts_salmon) == rownames(meta)))
ds_txi <- DESeqDataSetFromTximport(txi = txi_salmon,
                                   colData = meta,
                                   design = ~ group + flow)



```



In order to exclude genes with low expression, we use the edgeR package to filter the genes
Therefore, we create the DGE object from edgeR and normalize the read coverage of the genes

```{r,echo=FALSE , include = FALSE }

suppressPackageStartupMessages(library(edgeR))
genetable <- data.frame(gene.id = rownames(counts_salmon))
stopifnot(all(rownames(meta) == colnames(counts_salmon)))
dge <- DGEList(counts = counts_salmon, 
               samples = meta, 
               genes = genetable)

avetxlengths <- txi_salmon$length
stopifnot(all(rownames(avetxlengths) == rownames(counts_salmon)))
stopifnot(all(colnames(avetxlengths) == colnames(counts_salmon)))
avetxlengths <- avetxlengths/exp(rowMeans(log(avetxlengths)))
offsets <- log(calcNormFactors(counts_salmon/avetxlengths)) + 
  log(colSums(counts_salmon/avetxlengths))
dge$offset <- t(t(log(avetxlengths)) + offsets)
names(dge)


dge <- calcNormFactors(dge)
dge$samples
```


Next, the normalized counts from the edgeR package are used to filter low expressed genes, which would disturb the analysis.

```{r, echo=FALSE, include =FALSE }

## Calculate CPMs using the cpm() function from edgeR
cpms <- cpm(dge)
head(cpms)
keep_genes <- rowSums(cpms >= 0.1) >= 4
head(keep_genes)
table(keep_genes)

## Filter the DGEList
dge <- dge[keep_genes, ]

## Filter the DESeqDataSet to contain the same genes
ds_txi <- ds_txi[match(rownames(dge), rownames(ds_txi)), ]

ds_txi$group <- factor(ds_txi$group, levels=c("gn0", "gh0", "gf0", "gfl"))

```



For visual plotting of expression differences, we stabilize the varianz over the mean, using the variance stabilizing transformation (VST).

```{r, echo=FALSE, include =FALSE }

vsd <- vst(ds_txi)
head(colData(vsd),3)

```


Next, we perform a Principal Component Analysis (PCA) of the data in 3D and 2D.


```{r, echo=FALSE, include=FALSE }

vst <- assay(vsd)
colnames(vst) <- colnames(counts_salmon)

pc <- prcomp(t(vst))
percent <- round(summary(pc)$importance[2,]*100)
smpls <- conditions

```


We set color for the respective media group and plot the PCA

```{r, echo=FALSE, include=FALSE }

group.cols<-c("red","blue","green","black")
group.names<-c("gn0","gh0","gf0","gfl")

group <- factor(group, levels=c("gn0", "gh0", "gf0", "gfl"))

```


##### 3D:
```{r, echo=FALSE}
options(warn=-1)

suppressPackageStartupMessages(library(scatterplot3d))

scatterplot3d(pc$x[,1],
              pc$x[,2],
              pc$x[,3],
              xlab=paste("Comp. 1 (",percent[1],"%)",sep=""),
              ylab=paste("Comp. 2 (",percent[2],"%)",sep=""),
              zlab=paste("Comp. 3 (",percent[3],"%)",sep=""),
              color=group.cols[(factor(group))],
              pch=19)

legend("topright", inset=.03, title="Sample Group",
       levels(factor(group)), fill=c(group.cols), horiz=F, cex=0.8)

# library("rgl")
# library("car")
# scatter3d(x = pc$x[,1], y = pc$x[,2], z = pc$x[,3], groups = group,
#           surface=FALSE, ellipsoid = TRUE)

options(warn=0)

```



##### 2D:

```{r, echo=FALSE}

plot(pc$x[,1],
     pc$x[,2],
     xlab=paste("Comp. 1 (",percent[1],"%)",sep=""),
     ylab=paste("Comp. 2 (",percent[2],"%)",sep=""),
     col=group.cols[(factor(group))], 
     pch=19)

legend("topright", inset=0.001, title="Treatment",
       levels(factor(group)), fill=c(group.cols), horiz=F, cex=0.8)

```

To overlook the influence of the flow cell on the gene expression, we check if the samples are clusterd by the flow cell ID they were analysed in.

```{r, echo=FALSE}

plot(pc$x[,1],
     pc$x[,2],
     xlab=paste("Comp. 1 (",percent[1],"%)",sep=""),
     ylab=paste("Comp. 2 (",percent[2],"%)",sep=""),
     col=group.cols[(factor(flow))], 
     pch=19)

legend("topright", inset=0.001, title="Flowcell",
       levels(factor(flow)), fill=c(group.cols), horiz=F, cex=0.8)
```

The differences in the gene expression can not be explained by the respective flow cell of the samples. Every cluster shows 2 samples associated to on of both flow cells. 



After visualization of the read counts, we now want to do actual DGE comparing gene expression across samples:
  This function will print out a message for the various steps it performs. These are described in more detail in the manual page for the DESeq2 R package.

```{r, echo=FALSE}
options(warn=-1)
ds_txi <- DESeq(ds_txi)
#plotDispEsts(ds_txi)
options(warn=0)

```

Next, check the change in gene expression between two samples, comparing gf0 with gh0,  gfl with gf0 and gn0 with gh0

The analysis with DESeq2 was done for all sample comparisons and saved locally.


gf0 vs gh0
```{r,echo=FALSE}

res_gf0_vs_gh0 <- results(ds_txi,contrast=c("group","gf0","gh0"), alpha= 0.01)

res_gf0_vs_gh0_08_2FC <- results(ds_txi,contrast=c("group","gf0","gh0"),alpha= 0.01,lfcThreshold=.7)
summary(res_gf0_vs_gh0)


```
gfl vs gf0
```{r,echo=FALSE}

res_gfl_vs_gf0 <- results(ds_txi,contrast=c("group","gfl","gf0"), alpha= 0.01)
res_gfl_vs_gf0_08_2FC <- results(ds_txi,contrast=c("group","gfl","gf0"), alpha= 0.01,lfcThreshold=.7)
summary(res_gfl_vs_gf0)

```
gh0 vs gn0
```{r,echo=FALSE}

res_gh0_vs_gn0 <- results(ds_txi,contrast=c("group","gh0","gn0"), alpha= 0.01)
res_gh0_vs_gn0_08_2FC <- results(ds_txi,contrast=c("group","gh0","gn0"), alpha= 0.01,lfcThreshold=.7)
summary(res_gh0_vs_gn0)

```


We also test if the expression of the genes showes at least a +/- 0.7 two-fold change. Decreasing this value or increasing the adjsuted p-value from 1% to maybe 5% would lead to a higher number of detected differentially expressed genes.



```{r,echo=FALSE}

res_gfl_vs_gn0 <- results(ds_txi,contrast=c("group","gfl","gn0"), alpha= 0.01)

res_gfl_vs_gh0 <- results(ds_txi,contrast=c("group","gfl","gh0"), alpha= 0.01)



```



#### The MA-plot for the three gn0 comparison

The log2 fold change for a particular comparison is plotted on the y-axis and the average of the counts normalized by size factor is shown on the x-axis ("M" for minus, because a log ratio is equal to log minus log, and "A" for average). Each gene is represented with a dot. Genes with an adjusted p value below a threshold (here 0.1, the default with DESeq2) are shown in red.

gh0_vs_gn0
```{r}

DESeq2::plotMA(res_gh0_vs_gn0, ylim = c(-5, 5))
```


gfl_vs_gf0
```{r}

DESeq2::plotMA(res_gfl_vs_gf0, ylim = c(-5, 5))
```


gf0_vs_gh0
```{r}

DESeq2::plotMA(res_gf0_vs_gh0, ylim = c(-5, 5))
```

Looking at the different groups we can generate a heatmap of the 20 most variable expressed genes across samples.
Therefor the most 300 variable genes between every sample comparison are compared and the 20 most frequently occuring candidates are selected.

```{r,echo=FALSE}

res1 <- results(ds_txi, alpah=0.01, contrast= c("group","gfl","gn0"))
mat <- assay(vsd)[head(order(res1$padj,decreasing = F), 300), ]
mat <- mat - rowMeans(mat)
res1 <- rownames(mat)


res2 <- results(ds_txi, alpah=0.01, contrast= c("group","gfl","gh0"))
mat <- assay(vsd)[head(order(res2$padj,decreasing = F), 300), ]
mat <- mat - rowMeans(mat)
res2 <- rownames(mat)

res3 <- results(ds_txi, alpah=0.01, contrast= c("group","gfl","gf0"))
mat <- assay(vsd)[head(order(res3$padj,decreasing = F), 300), ]
mat <- mat - rowMeans(mat)
res3 <- rownames(mat)

res4 <- results(ds_txi, alpah=0.01, contrast= c("group","gf0","gn0"))
mat <- assay(vsd)[head(order(res4$padj,decreasing = F), 300), ]
mat <- mat - rowMeans(mat)
res4 <- rownames(mat)

res5 <- results(ds_txi, alpah=0.01, contrast= c("group","gf0","gh0"))
mat <- assay(vsd)[head(order(res5$padj,decreasing = F), 300), ]
mat <- mat - rowMeans(mat)
res5 <- rownames(mat)

res6 <- results(ds_txi, alpah=0.01, contrast= c("group","gh0","gn0"))
mat <- assay(vsd)[head(order(res6$padj,decreasing = F), 300), ]
mat <- mat - rowMeans(mat)
res6 <- rownames(mat)


ress <- c(res1,res2,res3,res4,res5,res6)
df <- data.frame(table(ress))
df <- df[order(df$Freq, decreasing = T),]
most_variant20 <- as.character(df[1:20,1])



```




```{r,echo=FALSE}

options(warn=-1)
suppressPackageStartupMessages(library(pheatmap))

rn_vsd <- row.names(vsd)
select <- match(most_variant20,rn_vsd)
mat <- assay(vsd)[select, ]
mat <- mat - rowMeans(mat)
df <- as.data.frame(colData(vsd)[, c("group")])
rownames(df) <- colnames(mat)

pheatmap(mat, annotation_col = df, cluster_rows = F, show_colnames = F )
options(warn=0)

```

Clustering the expression of these 20 genes by the samples treatment, lead to a distinct characteriuzation of the four different treatments.


The same can be done for the most unregulated genes:
  
  
  ```{r,echo=FALSE}

res1 <- results(ds_txi, alpah=0.01, contrast= c("group","gfl","gn0"))
mat <- assay(vsd)[head(order(res1$padj,decreasing = T), 300), ]
mat <- mat - rowMeans(mat)
res1 <- rownames(mat)


res2 <- results(ds_txi, alpah=0.01, contrast= c("group","gfl","gh0"))
mat <- assay(vsd)[head(order(res2$padj,decreasing = T), 300), ]
mat <- mat - rowMeans(mat)
res2 <- rownames(mat)

res3 <- results(ds_txi, alpah=0.01, contrast= c("group","gfl","gf0"))
mat <- assay(vsd)[head(order(res3$padj,decreasing = T), 300), ]
mat <- mat - rowMeans(mat)
res3 <- rownames(mat)

res4 <- results(ds_txi, alpah=0.01, contrast= c("group","gf0","gn0"))
mat <- assay(vsd)[head(order(res4$padj,decreasing = T), 300), ]
mat <- mat - rowMeans(mat)
res4 <- rownames(mat)

res5 <- results(ds_txi, alpah=0.01, contrast= c("group","gf0","gh0"))
mat <- assay(vsd)[head(order(res5$padj,decreasing = T), 300), ]
mat <- mat - rowMeans(mat)
res5 <- rownames(mat)

res6 <- results(ds_txi, alpah=0.01, contrast= c("group","gh0","gn0"))
mat <- assay(vsd)[head(order(res6$padj,decreasing = T), 300), ]
mat <- mat - rowMeans(mat)
res6 <- rownames(mat)


ress <- c(res1,res2,res3,res4,res5,res6)
df <- data.frame(table(ress))
df <- df[order(df$Freq, decreasing = T),]
most_variant20 <- as.character(df[1:20,1])



```



```{r,echo=FALSE}
suppressPackageStartupMessages(library(pheatmap))

rn_vsd <- row.names(vsd)
select <- match(most_variant20,rn_vsd)
mat <- assay(vsd)[select, ]
mat <- mat - rowMeans(mat)
df <- as.data.frame(colData(vsd)[, c("group")])
rownames(df) <- colnames(mat)

pheatmap(mat, annotation_col = df, cluster_rows = F, show_colnames = F  )

```

Here we see 20 genes that are not at all differentially expressed between the samples, resulting in the mosaik like image.



For better sample to sample comparison, we save the results from the DESeq2 calculation of expression differences and the respective adjusted p values.

Therefor, we need to add the GO terms to our respective results tables.
```{r, echo= FALSE}
gene_names_GOterms <- read.csv("S:/Arbeitsgruppen VIRO/AG_Sch/Johannes Ptok_JoPt/WHK/DEG/Salmon/gene_names_GOterms.txt", stringsAsFactors=FALSE)

```

This is an example of the content of the DGE tables.
```{r, echo = FALSE}

res_gfl_vs_gf0_ordered <- res_gfl_vs_gf0[order(res_gfl_vs_gf0$padj), ]
res_gfl_vs_gf0_ordered <- as.data.frame(res_gfl_vs_gf0_ordered)
res_gfl_vs_gf0_ordered$gene_name <- row.names(res_gfl_vs_gf0_ordered)
res_gfl_vs_gf0_ordered$gene_id <-  gene_names_GOterms$Gene.stable.ID[match(res_gfl_vs_gf0_ordered$gene_name, gene_names_GOterms$Gene.name)]
res_gfl_vs_gf0_ordered$GO_term <-  gene_names_GOterms$GO.term.accession[match(res_gfl_vs_gf0_ordered$gene_name, gene_names_GOterms$Gene.name)]
#res_gfl_vs_gf0_ordered <- res_gfl_vs_gf0_ordered[res_gfl_vs_gf0_ordered$padj <= 0.01 ,]


res_gfl_vs_gn0_ordered <- res_gfl_vs_gn0[order(res_gfl_vs_gn0$padj), ]
res_gfl_vs_gn0_ordered <- as.data.frame(res_gfl_vs_gn0_ordered)
res_gfl_vs_gn0_ordered$gene_name <- row.names(res_gfl_vs_gn0_ordered)
res_gfl_vs_gn0_ordered$gene_id <-  gene_names_GOterms$Gene.stable.ID[match(res_gfl_vs_gn0_ordered$gene_name, gene_names_GOterms$Gene.name)]
res_gfl_vs_gn0_ordered$GO_term <-  gene_names_GOterms$GO.term.accession[match(res_gfl_vs_gn0_ordered$gene_name, gene_names_GOterms$Gene.name)]
#res_gfl_vs_gn0_ordered <- res_gfl_vs_gn0_ordered[res_gfl_vs_gn0_ordered$padj <= 0.01 ,]




res_gfl_vs_gh0_ordered <- res_gfl_vs_gh0[order(res_gfl_vs_gh0$padj), ]
res_gfl_vs_gh0_ordered <- as.data.frame(res_gfl_vs_gh0_ordered)
res_gfl_vs_gh0_ordered$gene_name <- row.names(res_gfl_vs_gh0_ordered)
res_gfl_vs_gh0_ordered$gene_id <-  gene_names_GOterms$Gene.stable.ID[match(res_gfl_vs_gh0_ordered$gene_name, gene_names_GOterms$Gene.name)]
res_gfl_vs_gh0_ordered$GO_term <-  gene_names_GOterms$GO.term.accession[match(res_gfl_vs_gh0_ordered$gene_name, gene_names_GOterms$Gene.name)]
#res_gfl_vs_gh0_ordered <- res_gfl_vs_gh0_ordered[res_gfl_vs_gh0_ordered$padj <= 0.01 ,]



res_gh0_vs_gn0_ordered <- res_gh0_vs_gn0[order(res_gh0_vs_gn0$padj), ]
res_gh0_vs_gn0_ordered <- as.data.frame(res_gh0_vs_gn0_ordered)
res_gh0_vs_gn0_ordered$gene_name <- row.names(res_gh0_vs_gn0_ordered)
res_gh0_vs_gn0_ordered$gene_id <-  gene_names_GOterms$Gene.stable.ID[match(res_gh0_vs_gn0_ordered$gene_name, gene_names_GOterms$Gene.name)]
res_gh0_vs_gn0_ordered$GO_term <-  gene_names_GOterms$GO.term.accession[match(res_gh0_vs_gn0_ordered$gene_name, gene_names_GOterms$Gene.name)]
res_gh0_vs_gn0_ordered <- res_gh0_vs_gn0_ordered[res_gh0_vs_gn0_ordered$padj <= 0.01 ,]

res_gf0_vs_gh0_ordered <- res_gf0_vs_gh0[order(res_gf0_vs_gh0$padj), ]
res_gf0_vs_gh0_ordered <- as.data.frame(res_gf0_vs_gh0_ordered)
res_gf0_vs_gh0_ordered$gene_name <- row.names(res_gf0_vs_gh0_ordered)
res_gf0_vs_gh0_ordered$gene_id <-  gene_names_GOterms$Gene.stable.ID[match(res_gf0_vs_gh0_ordered$gene_name, gene_names_GOterms$Gene.name)]
res_gf0_vs_gh0_ordered$GO_term <-  gene_names_GOterms$GO.term.accession[match(res_gf0_vs_gh0_ordered$gene_name, gene_names_GOterms$Gene.name)]
res_gf0_vs_gh0_ordered <- res_gf0_vs_gh0_ordered[res_gf0_vs_gh0_ordered$padj <= 0.01,]

head(res_gf0_vs_gh0_ordered)




```

Finally we save the files
```{r, echo = FALSE}

write.csv2(res_gfl_vs_gf0_ordered , file = "gfl_vs_gf0.csv",  row.names = FALSE)
write.csv2(res_gh0_vs_gn0_ordered , file = "gh0_vs_gn0.csv",  row.names = FALSE)
write.csv2(res_gf0_vs_gh0_ordered , file = "gf0_vs_gh0.csv",  row.names = FALSE)

```



```{r, echo=FALSE, include=FALSE}

res_gfl_vs_gf0_ordered <- res_gfl_vs_gf0_08_2FC[order(res_gfl_vs_gf0_08_2FC$padj), ]
res_gfl_vs_gf0_ordered <- as.data.frame(res_gfl_vs_gf0_ordered)
res_gfl_vs_gf0_ordered$gene_name <- row.names(res_gfl_vs_gf0_ordered)
res_gfl_vs_gf0_ordered$gene_id <-  gene_names_GOterms$Gene.stable.ID[match(res_gfl_vs_gf0_ordered$gene_name, gene_names_GOterms$Gene.name)]
res_gfl_vs_gf0_ordered$GO_term <-  gene_names_GOterms$GO.term.accession[match(res_gfl_vs_gf0_ordered$gene_name, gene_names_GOterms$Gene.name)]
res_gfl_vs_gf0_ordered <- res_gfl_vs_gf0_ordered[res_gfl_vs_gf0_ordered$padj <= 0.01 & ((res_gfl_vs_gf0_ordered$log2FoldChange>= .7)|(res_gfl_vs_gf0_ordered$log2FoldChange<=-.7)),]

res_gh0_vs_gn0_ordered <- res_gh0_vs_gn0_08_2FC[order(res_gh0_vs_gn0_08_2FC$padj), ]
res_gh0_vs_gn0_ordered <- as.data.frame(res_gh0_vs_gn0_ordered)
res_gh0_vs_gn0_ordered$gene_name <- row.names(res_gh0_vs_gn0_ordered)
res_gh0_vs_gn0_ordered$gene_id <-  gene_names_GOterms$Gene.stable.ID[match(res_gh0_vs_gn0_ordered$gene_name, gene_names_GOterms$Gene.name)]
res_gh0_vs_gn0_ordered$GO_term <-  gene_names_GOterms$GO.term.accession[match(res_gh0_vs_gn0_ordered$gene_name, gene_names_GOterms$Gene.name)]
res_gh0_vs_gn0_ordered <- res_gh0_vs_gn0_ordered[res_gh0_vs_gn0_ordered$padj <= 0.01 & ((res_gh0_vs_gn0_ordered$log2FoldChange>= .7)|(res_gh0_vs_gn0_ordered$log2FoldChange<=-.7)),]

res_gf0_vs_gh0_ordered <- res_gf0_vs_gh0_08_2FC[order(res_gf0_vs_gh0_08_2FC$padj), ]
res_gf0_vs_gh0_ordered <- as.data.frame(res_gf0_vs_gh0_ordered)
res_gf0_vs_gh0_ordered$gene_name <- row.names(res_gf0_vs_gh0_ordered)
res_gf0_vs_gh0_ordered$gene_id <-  gene_names_GOterms$Gene.stable.ID[match(res_gf0_vs_gh0_ordered$gene_name, gene_names_GOterms$Gene.name)]
res_gf0_vs_gh0_ordered$GO_term <-  gene_names_GOterms$GO.term.accession[match(res_gf0_vs_gh0_ordered$gene_name, gene_names_GOterms$Gene.name)]
res_gf0_vs_gh0_ordered <- res_gf0_vs_gh0_ordered[res_gf0_vs_gh0_ordered$padj <= 0.01 & ((res_gf0_vs_gh0_ordered$log2FoldChange>= .7)|(res_gf0_vs_gh0_ordered$log2FoldChange<=-.7)) ,]

head(res_gf0_vs_gh0_ordered)


```


```{r, echo = FALSE}

write.csv2(res_gfl_vs_gf0_ordered , file = "gfl_vs_gf0_LFC_07.csv",  row.names = FALSE)
write.csv2(res_gh0_vs_gn0_ordered , file = "gh0_vs_gn0_LFC_07.csv",  row.names = FALSE)
write.csv2(res_gf0_vs_gh0_ordered , file = "gf0_vs_gh0_LFC_07.csv",  row.names = FALSE)

```


To check wethere the treatments show specific expression patterns, we cluster the samples by the expression of all genes.

```{r, echo = FALSE}


sampleDists <- dist(t(assay(vsd)))

library("RColorBrewer")
sampleDistMatrix <- as.matrix(sampleDists)
rownames(sampleDistMatrix) <- as.character( vsd$id)
colnames(sampleDistMatrix) <- NULL
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
pheatmap(sampleDistMatrix,
         clustering_distance_rows=sampleDists,
         clustering_distance_cols=sampleDists,
         col=colors)


```

Looking at the overall gene expression, we see distinct pattern in samples of the treatment group gfl. And a slight similarity of the groups gn0 and gh0.



#Visualization of gene expression accros samples

In order to get a first idea of the expression of certain genes across samples, we normalize the aligned reads by the library size and calcuate the standart derivation for every treatments average value.

```{r, echo = FALSE}
library(ggplot2)

ds_txi$group <- factor(ds_txi$group, levels=c("gn0", "gh0", "gf0", "gfl"))
gesamtDESeq <- counts(ds_txi, normalized=F)

counts_salmon_df <- data.frame(counts_salmon)
counts_salmon_df <- counts_salmon_df[,c(13:16,9:12,1:4,5:8)]
counts_salmon_df$gn <- row.names(counts_salmon_df)

vec <- numeric()
vec_col <- character()
  
for(i in c(1:16)){
  
  vec[i] <- sum(counts_salmon_df[,i])
  vec_col[i] <- names(counts_salmon_df)[i]
}

vv <- data.frame(treat=vec_col, total= vec)

make_gene_expression_barplot <- function(geneName){ 
  
  gene_selected <- geneName
  current <- gesamtDESeq[row.names(gesamtDESeq)== gene_selected,]
  current <- data.frame(current)
  current$sample <- row.names(current)
  current$treatment <- meta$group[match(current$sample,meta$id)]
  current$sample_total <- vv$total[match(current$sample,vv$treat)]
  
  current$current <- current$current/current$sample_total*1e6
  tm <- data.frame(tapply(current$current, current$treatment, mean))
  tm$name <- row.names(tm)
  
  current$treatment_mean <- tm[match(current$treatment,tm$name),1]
  current$treatment_mean <- as.numeric(current$treatment_mean)
  current$adj_cur <- current$current/unique(current$treatment_mean[current$treatment=="gn0"])
  current$treatment <- factor(current$treatment, levels=c("gn0", "gh0", "gf0", "gfl"))
  
  p <- ggplot(current, aes(x=treatment, y=adj_cur)) + stat_summary(fun.y = mean, geom="point") +
    geom_dotplot(binaxis='y', stackdir='center', dotsize=1)+
    ylab("Relative expression to gn0")+
     stat_summary(fun.data = mean_cl_normal, geom = "errorbar", width=0.2, size=1,color="red")+ theme(axis.title.x=element_blank(),axis.text.x = element_text( size=15),axis.text.y = element_text( size=15),axis.title.y= element_text( size=15) )
  
  
  
  return(p)
}

make_gene_expression_barplot("NOS3")

#plotCounts(ds_txi, "NOS3", intgroup = "group", normalized = F,xlab = "group")


```



```{r, echo = FALSE}

library(ggrepel)
library(ggplot2)
res_gh0_vs_gn0 <- results(ds_txi,contrast=c("group","gh0","gn0"), alpha= 0.01)
res_gf0_vs_gn0 <- results(ds_txi,contrast=c("group","gf0","gn0"), alpha= 0.01)
res_gfl_vs_gn0 <- results(ds_txi,contrast=c("group","gfl","gn0"), alpha= 0.01)



res_gh0_vs_gn0_ordered <- res_gh0_vs_gn0[order(res_gh0_vs_gn0$padj), ]
res_gh0_vs_gn0_ordered <- as.data.frame(res_gh0_vs_gn0_ordered)
res_gh0_vs_gn0_ordered$gene_name <- row.names(res_gh0_vs_gn0_ordered)
res_gh0_vs_gn0_ordered$gene_id <-  gene_names_GOterms$Gene.stable.ID[match(res_gh0_vs_gn0_ordered$gene_name, gene_names_GOterms$Gene.name)]
res_gh0_vs_gn0_ordered$GO_term <-  gene_names_GOterms$GO.term.accession[match(res_gh0_vs_gn0_ordered$gene_name, gene_names_GOterms$Gene.name)]


res_gf0_vs_gn0_ordered <- res_gf0_vs_gn0[order(res_gf0_vs_gn0$padj), ]
res_gf0_vs_gn0_ordered <- as.data.frame(res_gf0_vs_gn0_ordered)
res_gf0_vs_gn0_ordered$gene_name <- row.names(res_gf0_vs_gn0_ordered)
res_gf0_vs_gn0_ordered$gene_id <-  gene_names_GOterms$Gene.stable.ID[match(res_gf0_vs_gn0_ordered$gene_name, gene_names_GOterms$Gene.name)]
res_gf0_vs_gn0_ordered$GO_term <-  gene_names_GOterms$GO.term.accession[match(res_gf0_vs_gn0_ordered$gene_name, gene_names_GOterms$Gene.name)]



res_gfl_vs_gn0_ordered <- res_gfl_vs_gn0[order(res_gfl_vs_gn0$padj), ]
res_gfl_vs_gn0_ordered <- as.data.frame(res_gfl_vs_gn0_ordered)
res_gfl_vs_gn0_ordered$gene_name <- row.names(res_gfl_vs_gn0_ordered)
res_gfl_vs_gn0_ordered$gene_id <-  gene_names_GOterms$Gene.stable.ID[match(res_gfl_vs_gn0_ordered$gene_name, gene_names_GOterms$Gene.name)]
res_gfl_vs_gn0_ordered$GO_term <-  gene_names_GOterms$GO.term.accession[match(res_gfl_vs_gn0_ordered$gene_name, gene_names_GOterms$Gene.name)]



res_combi_gn0 <- res_gh0_vs_gn0_ordered
res_combi_gn0 <- res_combi_gn0[,c(1,7,8,9,2,5,6)]
names(res_combi_gn0) <- c("baseMean", "gene_name", "gene_id", "GO_term", "gh0_L2FC", "gh0_pvalue", "gh0_padj")

res_combi_gn0$gf0_L2FC <- res_gf0_vs_gn0_ordered$log2FoldChange[match(res_combi_gn0$gene_name, res_gf0_vs_gn0_ordered$gene_name)]
res_combi_gn0$gf0_pvalue <-res_gf0_vs_gn0_ordered$pvalue[match(res_combi_gn0$gene_name, res_gf0_vs_gn0_ordered$gene_name)]
res_combi_gn0$gf0_padj <-res_gf0_vs_gn0_ordered$padj[match(res_combi_gn0$gene_name, res_gf0_vs_gn0_ordered$gene_name)]


res_combi_gn0$gfl_L2FC <- res_gfl_vs_gn0_ordered$log2FoldChange[match(res_combi_gn0$gene_name, res_gfl_vs_gn0_ordered$gene_name)]
res_combi_gn0$gfl_pvalue <-res_gfl_vs_gn0_ordered$pvalue[match(res_combi_gn0$gene_name, res_gfl_vs_gn0_ordered$gene_name)]
res_combi_gn0$gfl_padj <-res_gfl_vs_gn0_ordered$padj[match(res_combi_gn0$gene_name, res_gfl_vs_gn0_ordered$gene_name)]


test <- (ds_txi[,c(13:16)])
rM <- rowMeans(counts(test, normalized=TRUE))
res_combi_gn0$baseMean_gn0 <- rM[match(res_combi_gn0$gene_name,names(rM))]

write.csv2(res_combi_gn0, file="Summary of DGE.csv",row.names=F)

make_gene_expression_barplot <- function(geneName){ 
  
  current <- res_combi_gn0[res_combi_gn0$gene_name == geneName,]
  current <- data.frame(treatment= c("gh0", "gf0", "gfl"), F2LC= c(current$gh0_L2FC, current$gf0_L2FC, current$gfl_L2FC), pvalue= c(current$gh0_pvalue, current$gf0_pvalue, current$gfl_pvalue))
  current$p_sig <- 0
  current$p_sig[current$pvalue<0.05] <- 1
  current$p_sig <- factor(current$p_sig)
  current$treatment <- factor(current$treatment, levels=c( "gh0", "gf0", "gfl"))
  
  
  #Create a custom color scale

myColors <- c("red","green")
names(myColors) <- levels(current$p_sig)
colScale <- scale_colour_manual(name = "p_sig",values = myColors)

current$textpvalue <- as.character(round(current$pvalue,2))
current$textpvalue <- paste0("p-value: ",current$textpvalue)

dis <- range(current$F2LC)[2]-range(current$F2LC)[1]/10

#One plot with all the data
p <- ggplot(current,aes(treatment,F2LC,label = current$textpvalue)) + geom_point()+ggtitle(paste0(geneName," - gn0 expression ", round(rM[geneName],1)))+ geom_text_repel() 
p <- p + colScale
  
return(p)
}

make_gene_expression_barplot("NOS3")

#plotCounts(ds_txi, "NOS3", intgroup = "group", normalized = F,xlab = "group")


```






NOS3/eNOS: control for disturbed endothelial cells
No significant change between gf0 and gfl detected



```{r, echo = FALSE}

#make_gene_expression_barplot("NOS3")

make_gene_expression_barplot("TXNIP")
#plotCounts(ds_txi, gene="TXNIP", intgroup="group", main="TXNIP")

```
TXNIP: Thioredoxin-interacting protein



```{r, echo = FALSE}
make_gene_expression_barplot("TXN")
#plotCounts(ds_txi, gene="TXN", intgroup="group", main="TXN/TRX")

```

TXN/TRX: Thioredoxin, response to intracellular nitric oxide



```{r, echo = FALSE}
make_gene_expression_barplot("TIMM23")
#plotCounts(ds_txi, gene="TIMM23", intgroup="group", main="TIMM23")

```

TIMM23: esential mitochondrial membran protein





```{r, echo = FALSE}
make_gene_expression_barplot("TOMM40")
#plotCounts(ds_txi, gene="TOMM40", intgroup="group", main="TOMM40")

```

TOMM40: Outerpart of mitochondrial pore







```{r, echo = FALSE}
#make_gene_expression_barplot("TERT")
#plotCounts(ds_txi, gene="TERT", intgroup="group", main="TERT")

```

TERT: Telomerase Reverse Transcriptase
to low expressed


```{r, echo = FALSE}
#make_gene_expression_barplot("GRHL3")
#plotCounts(ds_txi, gene="GRHL3", intgroup="group", main="GRHL3")

```

GRHL3: transcription factor, to stimulate migration of endothelial cells
to low expressed


```{r, echo = FALSE}
make_gene_expression_barplot("AKT2")
#plotCounts(ds_txi, gene="AKT2", intgroup="group", main="AKT2")

```
AKT2: responsible of the regulation of glucose uptake by mediating insulin-induced translocation of the SLC2A4/GLUT4 glucose transporter to the cell surface




```{r, echo = FALSE}
make_gene_expression_barplot("CDKN1B")
#plotCounts(ds_txi, gene="CDKN1B", intgroup="group", main="CDKN1B")

```

CDKN1B/p27: controls the cell cycle progression at G1



```{r, echo = FALSE}
make_gene_expression_barplot("TFAM")
#plotCounts(ds_txi, gene="TFAM", intgroup="group", main="TFAM")

```

TFAM: Transcription Factor A, Mitochondrial



```{r, echo = FALSE}
make_gene_expression_barplot("TFB1M")
#plotCounts(ds_txi, gene="TFB1M", intgroup="group", main="TFB1M")

```
TFB1M: Transcription Factor B1, Mitochondrial




```{r, echo = FALSE}
make_gene_expression_barplot("TFB2M")
#plotCounts(ds_txi, gene="TFB2M", intgroup="group", main="TFB2M")

```
TFB2M: Transcription Factor B2, Mitochondrial





```{r, echo = FALSE}
make_gene_expression_barplot("NR2F1")
#plotCounts(ds_txi, gene="NR2F1", intgroup="group", main="NR2F1")

```
NR2F1: nuclear hormone receptor and transcriptional regulator




```{r, echo = FALSE}
make_gene_expression_barplot("WRAP53")
#plotCounts(ds_txi, gene="WRAP53", intgroup="group", main="WRAP53/TCAB1")

```
WRAP53/TCAB1:  component of the telomerase holoenzyme complex




```{r, echo = FALSE}
make_gene_expression_barplot("DKC1")
#plotCounts(ds_txi, gene="DKC1", intgroup="group", main="DKC1")

```
DKC1: telomerase stabilization and maintenance




```{r, echo = FALSE}
make_gene_expression_barplot("PNPT1")
#plotCounts(ds_txi, gene="PNPT1", intgroup="group", main="PNPT1")

```
PNPT1: localized in the mitochondrial intermembrane space and is involved in import of RNA to mitochondria




```{r, echo = FALSE}
make_gene_expression_barplot("DIS3L2")
#plotCounts(ds_txi, gene="DIS3L2", intgroup="group", main="DIS3L2")

```
DIS3L2: Mediates degradation of uridylated pre-let-7 miRNAs, contributing to the maintenance of embryonic stem (ES) cells




```{r, echo = FALSE}
make_gene_expression_barplot("RNASE1")
#plotCounts(ds_txi, gene="RNASE1", intgroup="group", main="RNASE1")

```
RNASE1: cleavage of RNA on the 3' side of pyrimidine nucleotides




```{r, echo = FALSE}
make_gene_expression_barplot("RPP40")
#plotCounts(ds_txi, gene="RPP40", intgroup="group", main="RPP40")

```
RPP40: compount of RNASEP




```{r, echo = FALSE}
make_gene_expression_barplot("RPP30")
#plotCounts(ds_txi, gene="RPP30", intgroup="group", main="RPP30")

```
RPP30:  compount of RNASEP



```{r, echo = FALSE}
make_gene_expression_barplot("KIAA0391")
#plotCounts(ds_txi, gene="KIAA0391", intgroup="group", main="KIAA0391/MRPP3")

```
KIAA0391/MRPP3 :  component of mitochondrial ribonuclease P




```{r, echo = FALSE}
make_gene_expression_barplot("CXCR4")
#plotCounts(ds_txi, gene="CXCR4", intgroup="group", main="CXCR4")

```
CXCR4: regulating vascular branching and/or remodeling processes in endothelial cells




```{r, echo = FALSE}
make_gene_expression_barplot("HNRNPK")
#plotCounts(ds_txi, gene="HNRNPK", intgroup="group", main="HNRNPK")

```
HNRNPK: RNA binding protein and transcriptional coactivator


histone H2A

```{r, echo = FALSE}
make_gene_expression_barplot("SRSF3")
#plotCounts(ds_txi, gene="SRSF3", intgroup="group", main="SRSF3")

```
SRSF3: RNA binding protein


#Transkription

```{r, echo = FALSE}

#make_gene_expression_barplot("NOS3")

make_gene_expression_barplot("POLRMT")
#plotCounts(ds_txi, gene="TXNIP", intgroup="group", main="TXNIP")

```
MT polymerase needs TFAM and TFB2M for Transcription start. Both factors are down regulated
For elongation TEFM is needed




```{r, echo = FALSE}

#make_gene_expression_barplot("NOS3")

make_gene_expression_barplot("TEFM")
#plotCounts(ds_txi, gene="TXNIP", intgroup="group", main="TXNIP")

```


```{r, echo = FALSE}

#make_gene_expression_barplot("NOS3")

make_gene_expression_barplot("MTERF1")
#plotCounts(ds_txi, gene="TXNIP", intgroup="group", main="TXNIP")

```

MTERF1 



```{r, echo = FALSE}

#make_gene_expression_barplot("NOS3")

make_gene_expression_barplot("RPP25")
#plotCounts(ds_txi, gene="TXNIP", intgroup="group", main="TXNIP")

```
RNase P subunit p25 (in nucleus ?)







```{r, echo = FALSE}

#make_gene_expression_barplot("NOS3")

make_gene_expression_barplot("RPP21")
#plotCounts(ds_txi, gene="TXNIP", intgroup="group", main="TXNIP")

```

Ribonuclease P protein subunit p21 RPP21 (in nucleus?)






```{r, echo = FALSE}

#make_gene_expression_barplot("NOS3")

make_gene_expression_barplot("TRMT10C")
#plotCounts(ds_txi, gene="TXNIP", intgroup="group", main="TXNIP")

```
TRMT10C or  MRPP1, part of mitochondrial RNase P






```{r, echo = FALSE}

#make_gene_expression_barplot("NOS3")

make_gene_expression_barplot("HSD17B10")
#plotCounts(ds_txi, gene="TXNIP", intgroup="group", main="TXNIP")

```

HSD17B10 or MRPP2 part of mitochondrial RNase P








```{r, echo = FALSE}

#make_gene_expression_barplot("NOS3")

make_gene_expression_barplot("KIAA0391")
#plotCounts(ds_txi, gene="TXNIP", intgroup="group", main="TXNIP")

```


KIAA0391 or MRPP3  part of mitochondrial RNase P




```{r, echo = FALSE}

#make_gene_expression_barplot("NOS3")

make_gene_expression_barplot("ELAC2")
#plotCounts(ds_txi, gene="TXNIP", intgroup="group", main="TXNIP")

```
ELAC2 executes 3'-end maturation of both mitochondrial and nuclear pre-tRNAs


```{r, echo = FALSE}

#make_gene_expression_barplot("NOS3")

make_gene_expression_barplot("LRPPRC")
#plotCounts(ds_txi, gene="TXNIP", intgroup="group", main="TXNIP")

```
LRPPRC


stability of HSP-derived mitochondrial transcripts is regulated by leucine-rich penticopeptide rich domain containing protein (LRPPRC) positive effect on stability


```{r, echo = FALSE}

#make_gene_expression_barplot("NOS3")

make_gene_expression_barplot("PNPT1")
#plotCounts(ds_txi, gene="TXNIP", intgroup="group", main="TXNIP")

```
Knockdown of PNPase leads to the increase in the half-life of mitochondrial transcripts and the accumulation of RNA decay intermediates






```{r, echo = FALSE}

#make_gene_expression_barplot("NOS3")

make_gene_expression_barplot("ELAC2")
#plotCounts(ds_txi, gene="TXNIP", intgroup="group", main="TXNIP")

```
ELAC2

RNase Z






##FASTKDs

The 'tRNA punctuation model' does not explain all the primary transcript cleavage events, as not all mRNAs are immediately flanked by tRNAs. Various Fas-activated serine/threonine kinase (FASTK) proteins have been shown to be required for mtRNA stability and the processing of precursors, especially the non-canonical cleavage sites. They all contain a conserved nuclease fold (RAP domain); however, endonucleolytic activity has not been shown for any of the FASTK proteins [37]. For example, depletion or knockout of FASTKD2 leads to the accumulation of various cleavage precursors, especially 16 rRNA and ND6 mRNA [38,39]. Additionally, FASTK has been implicated in MT-ND6 maturation and stability, and FASTKD5, similar to FASTKD4, regulates the maturation of those precursor RNAs that cannot be processed by RNase P and ELAC2 [39,40]. Furthermore, cross-linking immunoprecipitation (CLIP)-based analysis of FADTKD2 binding sites identified 16S rRNA and ND6 as its targets [38]. Recently, FASTKD4 was shown to be required for the stable expression of several mt-mRNAs, whereas FASTKD1 had the opposite effect on the stability of the MT-ND3 mt-mRNA. Interestingly, depletion of both FASTKD1 and FASTKD4 also caused a loss of MT-ND3, suggesting that the loss of FASTKD4 is epistatic [37]. Moreover, FASTKD4 has been suggested to be responsible for the cleavage of the MT-ND5-CYB precursor. A detailed characterization of how the FASTK proteins regulate the mitochondrial transcriptome is likely to be a subject of intense study in the near future.

```{r, echo = FALSE}

#make_gene_expression_barplot("NOS3")

make_gene_expression_barplot("FASTKD2")
#plotCounts(ds_txi, gene="TXNIP", intgroup="group", main="TXNIP")

```


```{r, echo = FALSE}

#make_gene_expression_barplot("NOS3")

make_gene_expression_barplot("FASTKD1")
#plotCounts(ds_txi, gene="TXNIP", intgroup="group", main="TXNIP")

```


```{r, echo = FALSE}

#make_gene_expression_barplot("NOS3")

make_gene_expression_barplot("FASTKD5")
#plotCounts(ds_txi, gene="TXNIP", intgroup="group", main="TXNIP")

```



#maybe enhanced translation?

MTIF2 MTIF3 MTIF1 ?
  
  
  ```{r, echo = FALSE}

#make_gene_expression_barplot("NOS3")

make_gene_expression_barplot("MTIF2")
#plotCounts(ds_txi, gene="TXNIP", intgroup="group", main="TXNIP")

```



```{r, echo = FALSE}

#make_gene_expression_barplot("NOS3")

make_gene_expression_barplot("MTIF3")
#plotCounts(ds_txi, gene="TXNIP", intgroup="group", main="TXNIP")

```


```{r, echo = FALSE}

#make_gene_expression_barplot("NOS3")

make_gene_expression_barplot("UCP3")
#plotCounts(ds_txi, gene="TXNIP", intgroup="group", main="TXNIP")

```
UCP3 uncoppling protein 3, increases free radicals when upregulated



```{r, echo = FALSE}

#make_gene_expression_barplot("NOS3")

make_gene_expression_barplot("AKT3")
#plotCounts(ds_txi, gene="TXNIP", intgroup="group", main="TXNIP")

```
AKT3
Required for coordination of mitochondrial biogenesis with growth factor-induced increases in cellular energy demands.
Down-regulation by RNA interference reduces the expression of the phosphorylated form of BAD, resulting in the induction of caspase-dependent apoptosis



```{r, echo = FALSE}

#make_gene_expression_barplot("NOS3")

make_gene_expression_barplot("CDKN1A")
#plotCounts(ds_txi, gene="TXNIP", intgroup="group", main="TXNIP")

```

p53/TP53 mediated inhibition of cellular proliferation in response to DNA damage, blocking cell cycle progression


```{r, echo = FALSE}

#make_gene_expression_barplot("NOS3")

make_gene_expression_barplot("POLG")
#plotCounts(ds_txi, gene="TXNIP", intgroup="group", main="TXNIP")

```



```{r, echo = FALSE}

#make_gene_expression_barplot("NOS3")

make_gene_expression_barplot("PPARG")
#plotCounts(ds_txi, gene="TXNIP", intgroup="group", main="TXNIP")

```

```{r, echo = FALSE}

#make_gene_expression_barplot("NOS3")

make_gene_expression_barplot("POLG")
#plotCounts(ds_txi, gene="TXNIP", intgroup="group", main="TXNIP")

```

```{r, echo = FALSE}
test <- counts(ds_txi, normalized=F)
test <- data.frame(test)
test$gene_name <- row.names(test)
test$chr <- gene_names_GOterms$Chromosome.scaffold.name[match(test$gene_name,gene_names_GOterms$Gene.name)]

X <- split(test, test$chr)

chrom <- data.frame(chr= unique(test$chr) , gf0_1=0 ,gf0_2=0 ,gf0_3=0 ,gf0_4=0 ,gfl_1=0 ,gfl_2=0 ,gfl_3=0 ,gfl_4=0 ,  
                    gh0_1=0 ,gh0_2=0 ,gh0_3=0 ,gh0_4=0 ,gn0_1=0 ,gn0_2=0 ,gn0_3=0 ,gn0_4=0  )

chrom$chr <- as.character(chrom$chr)

for(i in 1:length(X)){
  
  cur <- X[[i]]
  chrom$chr[i] <- unique(cur$chr)
  chrom$gf0_1[i]<- sum(cur$gf0_1)
  chrom$gf0_2[i]<- sum(cur$gf0_2)
  chrom$gf0_3[i]<- sum(cur$gf0_3)
  chrom$gf0_4[i]<- sum(cur$gf0_4)
  
  chrom$gfl_1[i]<- sum(cur$gfl_1)
  chrom$gfl_2[i]<- sum(cur$gfl_2)
  chrom$gfl_3[i]<- sum(cur$gfl_3)
  chrom$gfl_4[i]<- sum(cur$gfl_4)
  
  chrom$gh0_1[i]<- sum(cur$gh0_1)
  chrom$gh0_2[i]<- sum(cur$gh0_2)
  chrom$gh0_3[i]<- sum(cur$gh0_3)
  chrom$gh0_4[i]<- sum(cur$gh0_4)
  
  chrom$gn0_1[i]<- sum(cur$gn0_1)
  chrom$gn0_2[i]<- sum(cur$gn0_2)
  chrom$gn0_3[i]<- sum(cur$gn0_3)
  chrom$gn0_4[i]<- sum(cur$gn0_4)
  
}
chrom2 <- chrom
chrom2[, ] <- lapply(chrom2[, ], as.character)

chrom2$chr <- as.character(chrom2$chr)

chrom5 <- chrom2[chrom2$chr %in% c("1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19",
                                   "20","21","22","X","Y"),]
chrom1 <- chrom2[!chrom2$chr %in% c("1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19",
                                    "20","21","22","X","Y","MT"),]
chrom4 <-  chrom2[chrom2$chr %in% "MT",]



chrom3 <- data.frame(t(chrom5))
test <- chrom3[1,]
row.names(test) <- NULL
colnames(test) <- NULL
test[, ] <- lapply(test[, ], as.character)
test <- as.character(test)

names(chrom3) <- test
chrom3<- chrom3[-1,]
chrom3$treatment <- row.names(chrom3)

write.table(chrom3, file="chromosome_sum_norm_coverage.csv",sep = ";", dec = ",", row.names = F)



chrom3 <- data.frame(t(chrom1))
test <- chrom3[1,]
row.names(test) <- NULL
colnames(test) <- NULL
test[, ] <- lapply(test[, ], as.character)
test <- as.character(test)

names(chrom3) <- test
chrom3<- chrom3[-1,]

chrom3$treatment <- row.names(chrom3)
write.table(chrom3, file="Unplaced_scaffold_sum_norm_coverage.csv",sep = ";", dec = ",", row.names = F)


chrom3 <- data.frame(t(chrom4))
test <- chrom3[1,]
row.names(test) <- NULL
colnames(test) <- NULL
test <- as.character(test)

names(chrom3) <- "MT"

chrom3$treatment <- row.names(chrom3)
chrom3 <- chrom3[chrom3$treatment!="chr",]
write.table(chrom3, file="MT_sum_norm_coverage.csv",sep = ";", dec = ",", row.names = F)


```
